src/Data/Mapper/PositionMapper.php

```diff
@@ -27,4 +27,24 @@ public function getPositions($options = [])
 
         return [$this->mapObjects($result), $paginate];
     }
+
+    protected function mapObjects(array $records)
+    {
+        $results = [];
+        foreach ($records as $record) {
+            $entity = new $this->entity;
```

Comment:
I think we could figure out something, the nice thing about using explicit setters and getters is that we know that no other part of the application is going to be able to change anything in the entity, unless it goes through that setter.